### DISCUSSION TITLE
Fresh PupilEXT setup in combo with an older remote eye-tracker for IR source

### DISCUSSION THREAD

**Discussion by kheki4** [openPupil/Open-PupilEXT#25] [Q&A] [answered]
Dear Mr. Babak Zandi and fellow researchers,

We have just received the necessary components yesterday and constructed a single camera setup to use with PupilEXT. The camera type we chose is a2A1920-160umBAS, and the lens is a VS Technologies SV-3514H. A 850mm IR-pass filter is also applied.

Ethical review is rather strict in our country, and my supervisor thinks that to use a custom made infrared light source would need to go through a complicated approval process. In order to get around this, me and my supervisor had an idea that our PupilEXT camera could exploit the IR lighting coming from our older remote eye tracker, an SMI Red 500. Do you think it is a good idea? At first glance, it seems to be working.

I attached a few pictures of the setup. (We are just about to make a proper box/spacer instead of the books under the monitor and the camera. Also the black camera holder is custom made for the PupilEXT camera.)

![DSCF9188](https://user-images.githubusercontent.com/33413678/187722574-8f188bd5-93cc-440b-8078-894a162622ed.JPG)
![DSCF9189](https://user-images.githubusercontent.com/33413678/187722588-a34a3fbd-e4ca-42f2-9117-28dc0caccf5d.JPG)
I accidentally forgot printscreens, but if you need I can grab them.

However we encountered a difficulty: Most of the time, the software detects the edge of the cornea instead of the pupil. Calibration with a checker pattern does finish successfully, but it does not seem to affect the said pupil detection problem.

What do you think could help?

Also, what gain and exposure settings do you recommend? Or how/on what scales do you recommend to determine them? For a quick test I used 3,00 for gain and 4000 for exposure, and image looked nice. How does this affect pupil detection?

Thank you in advance!

**Comment by BZandi** [comment, marked as answer]
**IR-light source:** Usually, the IR light source of such commercial eye tracking systems is driven with a lower duty cycle (PWM) frequency that matches the fps rate of the camera. Therefore, it could be possible that in PupilEXT, you record some dark images if you record an image during the "off-cycle" of the IR light. The live image view of PupilEXT is refreshed with 30 fps to be computer-recourse friendly, so you would not see this effect in the GUI. For this, you need to record only the camera's images with PupilEXT at the highest fps count of your camera using the blue dot in PupilEXT (see the video tutorials for the offline recording). Next, check the saved images on your local disk, whether there are dark images. If not, you can use the IR light source of SMI.

**Pupil detection quality:** It is essential to use the ROI functionality of PupilEXT to get the best results out of the pupil detection. For this, you need to click in the live view panel of PupilEXT on "ROI" and next on 0.3 or 0.6. Place the region of interest (ROI) over the eye's pupil. By using the ROI, PupilEXT will only iterate inside the selected area. However, in the next step, you need to activate the ROI. For this, you need to open the "Pupil Detection Settings" and activate the option "Use ROI Area selection". Furthermore, you need to adjust the parameters of the used algorithm. For example, for the PuRe algorithm (recommended), we have optimized the algorithm's parameters for different ROI sizes, which can be selected in the selection bar. Try the different available options and check whether the pupil detection accuracy can be enhanced. However, do not forget to press "Apply" after changing something in the "Pupil Detection Settings". If the edge of the cornea is recognized as a pupil, then you need to adjust the algorithm's parameters "Max Pupil Size" and "Mean Canthi Distance". With these parameters, you can enhance the pupil detection accuracy. These steps are necessary as the pupil detection algorithm's default parameters are optimized for a specific camera resolution and ratio of the eye size to image size, which is different in your case. Therefore parameter adjustments are necessary to get the best results.

https://user-images.githubusercontent.com/62992112/187913395-b42fa2ce-eab5-4069-a01a-4adf57f43d9a.mp4

I have attached a video, that will show you the "Pupil Detection Settings". On the right hand side in the "Pupil Detection Settings" window you can see the parameters of the PuRe algorithm.

**Gain/Exposure:** A high camera exposure value will limit you in adjusting higher fps counts. A lower fps count could lead to lower pupil detection accuracy during fast gaze jumps or eye blinks. In contrast, a high Gain could result in a bad signal-to-noise ratio in the image. Therefore, you need to find a good balance between these two values. It is advised to record offline images with PupilEXT and check the images manually with which options the best image quality can be reached.
